pkg/dns/dns.go

```diff
@@ -48,28 +49,38 @@ const (
 
 	// Resync period for the kube controller loop.
 	resyncPeriod = 30 * time.Minute
+
+	// Duration for which the TTL cache should hold the node.
+	nodeCacheTTL = 180 * time.Second
```

Comment:
Need to explain why this value was chosen please. What are the pros and cons of increasing or decreasing this value. Why is there a need to evict cache entries at all?  Are they not automatically kept up to date via watches?